src/main/java/redis/clients/jedis/Connection.java

```diff
@@ -334,4 +296,117 @@ protected Object readProtocolWithCheckingBroken() {
     }
     return responses;
   }
+
+  private static class DefaultJedisSocketFactory implements JedisSocketFactory {
```

Comment:
Shouldn't we set this class as public? So users can inherent and override if needed?

Reply:
I think of it as an implementation detail, users should create their own versions if they want something different, I am okay exposing it if there is a strong reasoning or you think it's mostly re-usable?

Reply:
Will extract.